message_ui.module

```diff
@@ -382,3 +365,29 @@ function message_ui_entity_load(array &$entities, $entity_type_id) {
     $entity->save();
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function message_ui_form_message_template_edit_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  /** @var \Drupal\message\MessageTemplateInterface $template */
+  $template = $form_state->getFormObject()->getEntity();
+  $form['third_party_settings'] = [
+    '#tree' => TRUE,
+    'message_ui' => [
+      'enabled' => [
+        '#type' => 'checkbox',
+        '#title' => t('Provide UI for this template'),
+        '#description' => t('Check this in order to provide user interface for this message template.'),
+        '#default_value' => $template->getThirdPartySetting('message_ui', 'enabled', FALSE),
```

Comment:
I think we should discuss whether the user interface should be enabled or disabled by default.

In our project we are already extensively using the Message stack for other purposes that do not require a UI, so for us it makes sense to have this disabled.

However I can imagine that for some other use cases it might make sense to have this enabled by default. For example if this is installed as a dependency of the Message Private module and the site builder does not use the Message stack for other purposes it would be preferable to have this enabled.

Reply:
Enabling the UI should be an explicit action, IMO.